iohk-monitoring/src/Cardano/BM/Backend/Switchboard.lhs

```diff
@@ -226,29 +244,55 @@ instance (FromJSON a, ToJSON a) => IsBackend Switchboard a where
         -- raises an exception, that exception will be re-thrown in the current
         -- thread, wrapped in ExceptionInLinkedThread.
         Async.linkOnly (not . isBlockedIndefinitelyOnSTM) dispatcher
-        putMVar sbref $ SwitchboardInternal {
-                            sbQueue = q,
-                            sbDispatch = dispatcher,
-                            sbLogBuffer = logbuf,
-                            sbLogBE = katipBE,
-                            sbBackends = bs }
+
+        -- Modify the internal state of the switchboard, the switchboard
+        -- is now running.
+        putMVar sbref $ SwitchboardInternal 
+            { sbQueue = q
+            , sbDispatch = dispatcher
+            , sbLogBuffer = logbuf
+            , sbLogBE = katipBE
+            , sbBackends = bs
+            , sbRunning = SwitchboardRunning
+            }
 
         return sb
 
     unrealize switchboard = do
-        let clearMVar :: MVar some -> IO ()
+        -- How is this "clearing" anything?
+        let clearMVar :: MVar (SwitchboardInternal a) -> IO ()
             clearMVar = void . tryTakeMVar
 
+        -- This is kind of weird. This function is generally used for
+        -- the modification of the internal state, not breaking the encapsulation.
         (dispatcher, queue) <- withMVar (getSB switchboard) (\sb -> return (sbDispatch sb, sbQueue sb))
+
         -- send terminating item to the queue
         lo <- LogObject <$> pure ["kill", "switchboard"]
                         <*> (mkLOMeta Warning Confidential)
                         <*> pure KillPill
+
         atomically $ TBQ.writeTBQueue queue lo
         -- wait for the dispatcher to exit
         res <- Async.waitCatch dispatcher
         either throwM return res
-        (clearMVar . getSB) switchboard
+
+        -- Let's switch the flag to not running. Yes, keep everything else the same.
+        let flagSwitchboardStopped :: SwitchboardInternal a -> SwitchboardInternal a
+            flagSwitchboardStopped sb = 
+                SwitchboardInternal 
+                    { sbQueue = sbQueue sb
+                    , sbDispatch = sbDispatch sb
+                    , sbLogBuffer = sbLogBuffer sb
+                    , sbLogBE = sbLogBE sb
+                    , sbBackends = sbBackends sb
+                    , sbRunning = SwitchboardNotRunning
+                    }
+
```

Comment:
why not: `flagSwitchboardStopped sb = sb { sbRunning = SwitchboardNotRunning }`

Reply:
really clearing/releasing the components?

Reply:
That is the same, but less explicit.